oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -157,12 +160,12 @@ public void destroy() throws Exception {
 	/**
 	 * Sets the {@link OAuth2AccessTokenResponseClient} to be used for getting an {@link OAuth2AuthorizedClient} for
 	 * client_credentials grant.
+	 * @deprecated Supply {@link #setClientAuthorizationProvider(OAuth2ClientAuthorizationProvider)} with a configured {@link OAuth2AccessTokenResponseClient} for the client_credentials grant.
 	 * @param clientCredentialsTokenResponseClient the client to use
 	 */
+	@Deprecated
 	public void setClientCredentialsTokenResponseClient(
 			OAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest> clientCredentialsTokenResponseClient) {
-		Assert.notNull(clientCredentialsTokenResponseClient, "clientCredentialsTokenResponseClient cannot be null");
```

Comment:
Since this is a WIP, I'm assuming this method just isn't filled out, yet. But, I'll call out that this method should not silently ignore the user's intent, even if it is deprecated. Instead, it should adapt to the new way or it should throw an exception that indicates it's no longer supported.